backend/app/mcp/tools.py

```
@@ -0,0 +1,606 @@
+
+        # generate_pdf 는 동기 메서드 — BytesIO 반환
+        pdf_io = _singletons.report_generator.generate_pdf(analysis_data)
+        pdf_bytes = pdf_io.getvalue()
+        size_bytes = len(pdf_bytes)
+
+        # 5MB 가드 — Design §12 리스크 (base64 페이로드 과대)
+        if size_bytes > 5 * 1024 * 1024:
+            raise ToolError(
+                f"PDF too large for inline base64 transport: {size_bytes} bytes (limit 5MB). "
+                f"Consider using GET /api/v1/reviews/report/pdf?analysis_id={record.id} instead."
+            )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)